src/main/java/com/microsoft/graph/serializer/ISerializer.java

```diff
@@ -36,25 +37,25 @@
     /**
      * Deserialize an object from the input string
      * 
-     * @param inputString the string that stores the representation of the item
+     * @param inputStream the string that stores the representation of the item
```

Comment:
exposing the overrides you have in the implementation that accept JsonElement would be really interesting as well.
That's because for a number of properties in the object model we do:
1. get the property node in the json document
2. toString()
3. pass it to the deserialize method
4. parse the string right back

which is a way of CPU cycles, especially when handling high volumes

Reply:
Yup, I can do that. It feels a little bit unusual though: The `ISerializer` appears to be a generic interface handling (de)serialization, but with the overrides we will make it clear it's about GSON.

Reply:
our models have Gson annotations, so we're deeply tied with Gson.

Reply:
Done, I added those methods to the interface.

Reply:
thanks, resolving this thread